File: dmn-test-cases/standard/tck/1.1/cl3/0031-static-user-defined-functions/translator/expected/java/dmn/type/TFnLibrary.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinitionInterface.ftl", "tFnLibrary"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(as = type.TFnLibraryImpl.class)
public interface TFnLibrary extends com.gs.dmn.runtime.DMNType {
    static TFnLibrary toTFnLibrary(Object other) {
        if (other == null) {
            return null;
        } else if (TFnLibrary.class.isAssignableFrom(other.getClass())) {
            return (TFnLibrary)other;
        } else if (other instanceof com.gs.dmn.runtime.Context) {
            TFnLibraryImpl result_ = new TFnLibraryImpl();
            Object sumFn = ((com.gs.dmn.runtime.Context)other).get("sumFn");
            result_.setSumFn((Object)sumFn);
            Object subFn = ((com.gs.dmn.runtime.Context)other).get("subFn");
            result_.setSubFn((Object)subFn);
            Object multiplyFn = ((com.gs.dmn.runtime.Context)other).get("multiplyFn");
            result_.setMultiplyFn((Object)multiplyFn);
            Object divideFn = ((com.gs.dmn.runtime.Context)other).get("divideFn");
            result_.setDivideFn((Object)divideFn);
            return result_;
        } else if (other instanceof com.gs.dmn.runtime.DMNType) {
            return toTFnLibrary(((com.gs.dmn.runtime.DMNType)other).toContext());
        } else {
            throw new com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.getClass().getSimpleName(), TFnLibrary.class.getSimpleName()));
        }
    }

    @com.fasterxml.jackson.annotation.JsonGetter("sumFn")
    Object getSumFn();

    @com.fasterxml.jackson.annotation.JsonGetter("subFn")
    Object getSubFn();

    @com.fasterxml.jackson.annotation.JsonGetter("multiplyFn")
    Object getMultiplyFn();

    @com.fasterxml.jackson.annotation.JsonGetter("divideFn")
    Object getDivideFn();

    default com.gs.dmn.runtime.Context toContext() {
        com.gs.dmn.runtime.Context context = new com.gs.dmn.runtime.Context();
        context.add("sumFn", getSumFn());
        context.add("subFn", getSubFn());
        context.add("multiplyFn", getMultiplyFn());
        context.add("divideFn", getDivideFn());
        return context;
    }

    default boolean equalTo(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TFnLibrary other = (TFnLibrary) o;
        if (this.getDivideFn() != null ? !this.getDivideFn().equals(other.getDivideFn()) : other.getDivideFn() != null) return false;
        if (this.getMultiplyFn() != null ? !this.getMultiplyFn().equals(other.getMultiplyFn()) : other.getMultiplyFn() != null) return false;
        if (this.getSubFn() != null ? !this.getSubFn().equals(other.getSubFn()) : other.getSubFn() != null) return false;
        if (this.getSumFn() != null ? !this.getSumFn().equals(other.getSumFn()) : other.getSumFn() != null) return false;

        return true;
    }

    default int hash() {
        int result = 0;
        result = 31 * result + (this.getDivideFn() != null ? this.getDivideFn().hashCode() : 0);
        result = 31 * result + (this.getMultiplyFn() != null ? this.getMultiplyFn().hashCode() : 0);
        result = 31 * result + (this.getSubFn() != null ? this.getSubFn().hashCode() : 0);
        result = 31 * result + (this.getSumFn() != null ? this.getSumFn().hashCode() : 0);
        return result;
    }

    default String asString() {
        StringBuilder result_ = new StringBuilder("{");
        result_.append("divideFn=" + getDivideFn());
        result_.append(", multiplyFn=" + getMultiplyFn());
        result_.append(", subFn=" + getSubFn());
        result_.append(", sumFn=" + getSumFn());
        result_.append("}");
        return result_.toString();
    }
}

File: dmn-test-cases/standard/tck/1.1/cl3/0031-static-user-defined-functions/translator/expected/kotlin/dmn/type/TFnLibrary.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinitionInterface.ftl", "tFnLibrary"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(`as` = type.TFnLibraryImpl::class)
interface TFnLibrary : com.gs.dmn.runtime.DMNType {
    @get:com.fasterxml.jackson.annotation.JsonGetter("sumFn")
    val sumFn: kotlin.Any?

    @get:com.fasterxml.jackson.annotation.JsonGetter("subFn")
    val subFn: kotlin.Any?

    @get:com.fasterxml.jackson.annotation.JsonGetter("multiplyFn")
    val multiplyFn: kotlin.Any?

    @get:com.fasterxml.jackson.annotation.JsonGetter("divideFn")
    val divideFn: kotlin.Any?

    override fun toContext(): com.gs.dmn.runtime.Context {
        val context = com.gs.dmn.runtime.Context()
        context.put("sumFn", this.sumFn)
        context.put("subFn", this.subFn)
        context.put("multiplyFn", this.multiplyFn)
        context.put("divideFn", this.divideFn)
        return context
    }

    fun equalTo(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as TFnLibrary
        if (if (this.divideFn != null) this.divideFn != other.divideFn else other.divideFn != null) return false
        if (if (this.multiplyFn != null) this.multiplyFn != other.multiplyFn else other.multiplyFn != null) return false
        if (if (this.subFn != null) this.subFn != other.subFn else other.subFn != null) return false
        if (if (this.sumFn != null) this.sumFn != other.sumFn else other.sumFn != null) return false

        return true
    }

    fun hash(): Int {
        var result = 0
        result = 31 * result + (if (this.divideFn != null) this.divideFn.hashCode() else 0)
        result = 31 * result + (if (this.multiplyFn != null) this.multiplyFn.hashCode() else 0)
        result = 31 * result + (if (this.subFn != null) this.subFn.hashCode() else 0)
        result = 31 * result + (if (this.sumFn != null) this.sumFn.hashCode() else 0)
        return result
    }

    fun asString(): String {
        val result_ = StringBuilder("{")
        result_.append("divideFn=" + divideFn)
        result_.append(", multiplyFn=" + multiplyFn)
        result_.append(", subFn=" + subFn)
        result_.append(", sumFn=" + sumFn)
        result_.append("}")
        return result_.toString()
    }

    companion object {
        @JvmStatic
        fun toTFnLibrary(other: Any?): TFnLibrary? {
            if (other == null) {
                return null
            } else if (other is TFnLibrary?) {
                return other
            } else if (other is com.gs.dmn.runtime.Context) {
                var result_ = TFnLibraryImpl()
                result_.sumFn = other.get("sumFn") as kotlin.Any?
                result_.subFn = other.get("subFn") as kotlin.Any?
                result_.multiplyFn = other.get("multiplyFn") as kotlin.Any?
                result_.divideFn = other.get("divideFn") as kotlin.Any?
                return result_
            } else if (other is com.gs.dmn.runtime.DMNType) {
                return toTFnLibrary(other.toContext())
            } else {
                throw com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.javaClass.getSimpleName(), TFnLibrary::class.java.getSimpleName()))
            }
        }
    }
}
